graphql-tools-mocking/index.ts

```diff
@@ -35,8 +37,68 @@ const schemaString = `
   }
 `;
 
+let id = 1;
+
+function idGenerator() {
+  return id++;
+}
+
+function urnDirective(schema: GraphQLSchema): GraphQLSchema {
+  const directiveName = 'urn';
+
+  return mapSchema(schema, {
+    [MapperKind.OBJECT_FIELD]: (fieldConfig) => {
+      const urnDirectiveArgs = getDirective(
+        schema,
+        fieldConfig,
+        directiveName
+      )?.[0] as
+        | {
+            type: string;
+            fields?: [];
+          }
+        | undefined;
+
+      if (urnDirectiveArgs) {
+        const { resolve = defaultFieldResolver } = fieldConfig;
+
+        return {
+          ...fieldConfig,
+          resolve: async function (source, args, context, info) {
+            const result = await resolve(source, args, context, info);
+
+            if (typeof result === 'string') {
+              const { type, fields = [] } = urnDirectiveArgs;
+
+              const baseUrn = `unique_urn_with_parts_${type}`;
+
+              if (fields.length) {
+                return `${baseUrn}:(${fields.map(() => idGenerator())})`;
+              }
+
+              return baseUrn;
+            }
```

Comment:
Looks good for our purposes, but WDYT about having the pattern be regex based so it's more widely applicable than just URNs?